services/ui-src/src/utils/other/parsing.test.tsx

```diff
@@ -58,3 +95,14 @@ describe("Test parseCustomHtml", () => {
     expect(sanitizationSpy).toHaveBeenCalled();
   });
 });
+
+describe("Test labelTextWithOptional", () => {
```

Comment:
Request: Right now this test is pulling in the enter component and report context, etc. and testing the utility where it's used in components. It would be better to move a testing like this to `TextField.test.tsx` where we already have all this mocked out and where we are testing the rest of the component. 

Here in `parsing.test.tsx`, we should just test the `labelTextWithOptional` method itself. We could do that by testing the input and output of the function. So something like: if you pass a label string of "test value", you get back `<span>test value <span class='optional-text'> (optional)</span></span>` or whatever comes back. 

Changing the tests like this helps keep concerns separate. The utility test file just tests the utilities and the component test file just tests the component.